src/coreclr/utilcode/md5.cpp

```diff
@@ -141,11 +142,12 @@ void MD5::GetHashValue(MD5HASHDATA* phash)
     //
     // but our compiler has an intrinsic!
 
-    #if (defined(HOST_X86) || defined(HOST_ARM) || !defined(__clang__)) && defined(TARGET_UNIX)
-    #define ROL(x, n)        (((x) << (n)) | ((x) >> (32-(n))))
-    #define ROTATE_LEFT(x,n) (x) = ROL(x,n)
+    #if defined(TARGET_WINDOWS)
+        #define ROTATE_LEFT(x, n) (x) = _lrotl(x, n)
+    #elif __has_builtin(__builtin_rotateleft)
+        #define ROTATE_LEFT(x, n) ((x) = __builtin_rotateleft(x, n))
     #else
-    #define ROTATE_LEFT(x,n) (x) = _lrotl(x,n)
+        #define ROTATE_LEFT(x, n) ((x) = ((x << (n)) | (x >> ((sizeof(x) * 8 - 1) & -(n)))))
```

Comment:
I wonder what is the benefit of using `((sizeof(x) * 8 - 1) & -(n)))` as a shift amount instead of an easier to read `(sizeof(x) * 8 - (n))`.

Reply:
```suggestion
        #define ROTATE_LEFT(x, n) ((x) = ((x << (n)) | (x >> (sizeof(x) * 8 - (n)))))
```